numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -1528,7 +1533,13 @@ mapiter_trivial_@name@(PyArrayObject *self, PyArrayObject *ind,
             assert(npy_is_aligned(self_ptr, _UINT_ALIGN(@copytype@)));
             *(@copytype@ *)result_ptr = *(@copytype@ *)self_ptr;
 #else
-            copyswap(result_ptr, self_ptr, 0, self);
```

Comment:
Out of curiosity, would leaving in an `object`-specific implementation that still does `copyswap` get back some of the performance loss? Or is it all in the overhead of setting up the `cast_info` struct?